### PR TITLE
Bump to golang v1.18.3

### PR DESCRIPTION
**Pull request by weihanglo**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**
Golang version upgrade

**Solution:**
Bump to golang 1.18.2 (other deps would be upgraded in follow-ups)

**Related Issue:**
#2267 #2315

**Test plan:**

#### Build an ISO

1. Build the ISO
2. Do some manual smoke tests against the ISO

#### Validate you can develop harvester locally

1. `make build`
2. `make package`
3. The image is built successfully and can be use in a existing cluster.

### CODE REVIEW

**Comment by weihanglo** [comment]
I believe that we can upgrade golang itself and other dependencies separately based on Go's [promise of compatibility](https://go.dev/doc/go1compat). The ISO work as expected to boostrap a cluster. If the team feel uncomfortable and want to bump deps in one PR, I can do that also.

**Review by guangbochen — APPROVED** [review]
LGTM and we can manage the other dependency bump separately via https://github.com/harvester/harvester/issues/2315, thanks.

**Comment by guangbochen** [comment]
btw @weihanglo, with checking the [golang's EOL](https://endoflife.date/go), I'm thinking we can consider bumping to go v1.18 with Go 1 promise of compatibility support and new language enhancement.

File: Dockerfile.dapper [hunk 1]
```diff
@@ -19,7 +19,7 @@ RUN apt-get update -qq && apt-get install -y --no-install-recommends \
     && rm -rf /var/lib/apt/lists/*
 
 # install yq
-RUN GO111MODULE=on go get github.com/mikefarah/yq/v4@v4.6.0
+RUN GO111MODULE=on go install github.com/mikefarah/yq/v4@v4.6.0
```
Comment (weihanglo) [inline Dockerfile.dapper:22]: From 1.18, `go install` is the favoured command to install executable.

Ref: https://go.dev/doc/go1.18#go-get

**Comment by weihanglo** [comment]
@guangbochen Just updated to Golang v1.18. Thanks for the suggestion!

**Review by guangbochen — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -101,10 +99,179 @@ require (
...
-	kubevirt.io/client-go v0.49.0 // indirect
 	kubevirt.io/containerized-data-importer-api v1.41.0
 	kubevirt.io/kubevirt v0.49.0
 	sigs.k8s.io/cluster-api v0.4.4
 	sigs.k8s.io/kind v0.11.1
 	sigs.k8s.io/yaml v1.2.0
 )
+
+require (
+	github.com/Azure/go-ansiterm v0.0.0-20170929234023-d6e3b3328b78 // indirect
```
Comment (weihanglo) [inline go.mod:110]: I believe this change is the result of module graph pruning from 1.17
https://go.dev/doc/go1.17#graph-pruning

**Comment by weihanglo** [comment]
@guangbochen 
1.18.3 was released 5 days ago. Including some security fix might be better to bump. I updated this PR and request your re-review, thanks.
https://go.dev/doc/devel/release#go1.18.minor

**Review by bk201 — APPROVED** [review]
LGTM! we should be able to remove `GO111MODULE=on` in the future since it's not needed.

**Review by guangbochen — APPROVED** [review]

**Pull request merged**